Container-License-Compliance/containers.md

```diff
@@ -0,0 +1,108 @@
+
+Pushing an Image
+: The act of 'pushing' an Image to a Registry. In reality, all the layers the Image is composed of are pushed to the Registry (for verification), and only new layers that do not already exist on the Registry are updated and stored there
+
+Registry
+: A server/service used to upload, store and retrieve container Images. In reality, a registry works with and stores Layers
+
+Tarball (Image)
+: A single tar file comprising of all the layers that make up an Image. Generated with a 'docker save'
+
+Tarball (container)
+: A single tar file comprising of all the layers that make up a 'Container', squashed into a single Layer. Generated with a 'docker export'
+
+## More information
+
+There are usually three types of container-related deliveries.
+
+### Dockerfiles
+
+Dockerfiles are text files that describe how to build container images. They do not themselves contain any other software but instruct the building tooling to fetch software and/or manipulate the existing configuration to produce an image.
+
+The following example Dockerfile would produce a Clear Linux based Linux image with the git tool installed:
+
+FROM clearlinux:base
+RUN swupd update && swupd bundle-add git
+
+### Container Image (layers)
+
+The container image layers composited from a Dockerfile. The image is composed of one or more layers, and of particular interest is what layers you have added, and which, if any, layers (images) you have based your Image upon (the FROMs in your Dockerfile).
+
+### Container tarfile Images
+
+These are generated from a docker 'export' or 'save'. A full disk image of a container including all the components (layers).  Not to be confused with the executable container images produced locally by the tooling for local container execution.
+
+
+## Things to be taken care of, in order to get approval to deliver
+
+There are usually three types of container-related deliveries.
+
+### Dockerfiles
+
+You will need to ensure that your Dockerfile scripts are security reviewed, that they fetch current versions of software securely from trusted sources, and that you have mechanisms in place to update them as needed if for example it becomes necessary to use new versions of packages due to security errors.
+
+As far as possible you should pull packages from the base distribution itself using the distribution installation tools (e.g., apt, dnf, yum or swupd etc.). Where you cannot do this, you should pull from a trusted source, e.g., an internal build that already has approval. That pull should be via https:// or similar mechanisms and if appropriate also check signatures.
+
+You must list your Dockerfile (in the created/modified components table), along with any software that your Dockerfile copies or installs into your layers (in the unmodified libraries table).
```

Comment:
What do you mean with this?

Reply:
Too specific: our approval request process asks people to fill two tables (BOM), one for modified components and one for unmodified ones. I'll re-word it, but obviously others might follow different processes.